### DISCUSSION TITLE
How To Configure cypress-grep plugin v3.1.0 for Cypress v11.01 (TS not JS)

### DISCUSSION THREAD

**Discussion by Zujaj** [cypress-io/cypress#24639] [Plugins]
Thanks @bahmutov for creating the [@cypress-grep](https://github.com/cypress-io/cypress/tree/develop/npm/grep#cypressgrep) plugin. It's really cool and awesome, I saw the videos and appreciate your efforts. Here's the scenario:

### Cypress Version
11.0.1

### Node Version
16.16.0

### Package Manager
Yarn version 1.22.19

### Operating System
Windows 10 Pro Version 21H2

I've created a [cypress_grep_test_project](https://github.com/Zujaj/cypress_grep_test_project) to practice this plugin(version ^3.1.0) and used the sample spec files that Cypress provides during the first-time configuration. Just changed some files from "cy.js" to "cy.ts." I aim to use this plugin and test it out for Continuous Integration.

According to the [Cypress docs](https://docs.cypress.io/guides/tooling/plugins-guide#Using-a-plugin), the plugins folder no longer exists. Following the [current documentation of the plugin](https://github.com/cypress-io/cypress/tree/develop/npm/grep#support-file), the `registerCypressGrep` method isn't available anymore. 

I've added the `@cypress/grep` to the `ts.config` file. 
```ts
{
  "compilerOptions": {
    "target": "ES2022",
    "module": "commonjs",
    "types": [
      "cypress",
      "@cypress/grep" // Added this to prevent tags issue
    ],
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true
  }
}
```

In the `todo.cy.ts` file, I inserted the `{ tags: 'smoke' }` to the `describe` block as shown in below snippet.

```ts
describe('example to-do app', { tags: 'smoke' }, () => {
  beforeEach(() => {
    cy.visit('https://example.cypress.io/todo')
  })
```

Now I don't know what to do next, can you please help me with this? 

```ts
// cypress.config.ts
import { defineConfig } from 'cypress'

export default defineConfig({
  e2e: {
    video: false,
    setupNodeEvents(on, config) {
      // implement node event listeners here
    // What next?
    },
  }
})
```

**Comment by Zujaj** [comment]
Thanks @mahadazad for your collaboration.

Although this is just a workaround, I hope we'll get a proper solution to this problem very soon.

1. Navigate to the **`cypress\support\e2e.ts`** file.

![Navigation To e2e.ts file](https://user-images.githubusercontent.com/46039324/201313339-318060cf-d4c6-44ad-94c3-844178db8589.png)

2. Add the below code with `@ts-ignore` to ignore typescript compiler warnings.

```ts
// e2e.ts file

import './commands'

// @ts-ignore
import registerCypressGrep from '@cypress/grep' 

registerCypressGrep()
```
3. Run the command `npx cypress run --env grepTags=smoke` on the terminal.

![Result](https://user-images.githubusercontent.com/46039324/201315414-56ecb80e-0190-4d55-b241-f8d38eb4275c.png)

**Comment by T-e-j-k-u-m-a-r** [comment]
Hi @Zujaj : I tried the same steps suggested by you and getting below error and all the specs are getting executed: 

![image](https://github.com/cypress-io/cypress/assets/54025065/395ea356-b368-4152-89f5-1c9963051e99)


cypress.config.js file: 
![image](https://github.com/cypress-io/cypress/assets/54025065/d795cba7-603e-4584-bee4-dcbbf5a1b919)


spec file: 
![image](https://github.com/cypress-io/cypress/assets/54025065/f2bc2873-45e4-4430-8ac1-79f9985f9690)

e2e.ts file: 
![image](https://github.com/cypress-io/cypress/assets/54025065/aa42d7cb-0228-487f-b901-7fff15904eb9)

Execution report: 
Its should have executed only the 1st spec file which has the smoke tag: SE-2949-TC1, But instead it executed all the 4 spec files inside the login module. 

![image](https://github.com/cypress-io/cypress/assets/54025065/d4a8e770-7831-42d7-8017-ab750d9042fc)


![image](https://github.com/cypress-io/cypress/assets/54025065/d4a8e770-7831-42d7-8017-ab750d9042fc)


Can you please assist?

**Reply by Zujaj** [reply]
Hi @T-e-j-k-u-m-a-r 

I don't think you've followed the steps correctly. The e2e image you sent refers to your cypress test **SE-2949_TC1.cy.ts** file. I don't know what you've coded in the **e2e.ts**. I would suggest you revisit the solution by cloning the [cypress_grep_test_project ](https://github.com/Zujaj/cypress_grep_test_project) repo on your machine & compare the files side by side. 

PS: Please share the code rather than the images.

**Reply by Zujaj** [reply]
You can also read the [Typescript support section](https://github.com/cypress-io/cypress/tree/develop/npm/grep#typescript-support) added to the grep documentation.

**Comment by T-e-j-k-u-m-a-r** [comment]
Error message: 

![Uploading image.png…]()
